File: server/datasets/python_source-code/Multiple-G1/renamed_36_bst.py
class Node_0:
    def __init__(self, val_0):
        self.val_0 = val_0
        self.lhs_0 = None
        self.rhs_0 = None

class Tree_0:
    def __init__(self):
        self.rt_0 = None

    def ins_0(self, val_0):
        if self.rt_0 is None:
            self.rt_0 = Node_0(val_0)
        else:
            self._ins_0_rec(self.rt_0, val_0)

    def _ins_0_rec(self, nd_0, val_0):
        if val_0 < nd_0.val_0:
            if nd_0.lhs_0 is None:
                nd_0.lhs_0 = Node_0(val_0)
            else:
                self._ins_0_rec(nd_0.lhs_0, val_0)
        elif val_0 > nd_0.val_0:
            if nd_0.rhs_0 is None:
                nd_0.rhs_0 = Node_0(val_0)
            else:
                self._ins_0_rec(nd_0.rhs_0, val_0)

    def srch_0(self, val_0):
        return self._srch_0_rec(self.rt_0, val_0)

    def _srch_0_rec(self, nd_0, val_0):
        if nd_0 is None or nd_0.val_0 == val_0:
            return nd_0
        if val_0 < nd_0.val_0:
            return self._srch_0_rec(nd_0.lhs_0, val_0)
        return self._srch_0_rec(nd_0.rhs_0, val_0)

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_g1/renamed_36_bst.java
class Node_0 {
    int val_0;
    Node_0 lhs_0, rhs_0;
    public Node_0(int item) {
        val_0 = item;
        lhs_0 = rhs_0 = null;
    }
}
public class Tree_0 {
    public static void main(String[] args) {
        Tree_0 bst = new Tree_0();
        bst.ins_0(50);
        bst.srch_0(50);
    }
    Node_0 rt_0;
    public void ins_0(int val_0) {
        rt_0 = in_rec_0(rt_0, val_0);
    }
    Node_0 in_rec_0(Node_0 rt_0, int val_0) {
        if (rt_0 == null) return new Node_0(val_0);
        if (val_0 < rt_0.val_0) rt_0.lhs_0 = in_rec_0(rt_0.lhs_0, val_0);
        else if (val_0 > rt_0.val_0) rt_0.rhs_0 = in_rec_0(rt_0.rhs_0, val_0);
        return rt_0;
    }
    Node_0 srch_0(int val_0) {
        return sr_rec_0(rt_0, val_0);
    }
    Node_0 sr_rec_0(Node_0 rt_0, int val_0) {
        if (rt_0 == null || rt_0.val_0 == val_0) return rt_0;
        if (rt_0.val_0 < val_0) return sr_rec_0(rt_0.rhs_0, val_0);
        return sr_rec_0(rt_0.lhs_0, val_0);
    }
}